src/transformers/models/deberta/modeling_deberta.py

```diff
@@ -777,25 +773,6 @@ def _init_weights(self, module):
             if module.padding_idx is not None:
                 module.weight.data[module.padding_idx].zero_()
 
-    def _pre_load_hook(self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs):
```

Comment:
This logic is not necessary anymore. This is a slightly breaking change though since users will now have to add the `ingore_mismatched_sizes=True` flag to have this work.

Reply:
That preload hook was set up here in order to do exactly what your PR is doing, so it's fine to remove it here.